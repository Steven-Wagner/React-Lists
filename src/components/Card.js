import React from 'react';
import './Card.css';

class Card extends React.Component {
    render() {
        return (
        <div className="Card">
            <h3>{this.props.title}</h3>
            <p>{this.props.content}</p>
            <button onClick={()=>this.props.onDeleteItem(this.props.id)}>Delete</button>
        </div>
        );
    } 
}

export default Card;