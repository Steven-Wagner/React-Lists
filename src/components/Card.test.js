import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card';

describe('cards component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div);
        });

    it('renders the UI as expected', () => {
        const tree = renderer
        .create(<Card title='first card'
                    content='lorum ibsum'
                    key='a'/>)
        .toJSON();
        expect(tree).toMatchSnapshot();  
    });
})