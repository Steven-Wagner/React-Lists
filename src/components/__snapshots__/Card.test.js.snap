// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cards component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <h3>
    first card
  </h3>
  <p>
    lorum ibsum
  </p>
</div>
`;
